napari_assistant/_gui/_Assistant.py

```diff
@@ -465,3 +473,48 @@ def search_biii(self):
 def _open_url(url):
     import webbrowser
     webbrowser.open(url, new=0, autoraise=True)
+
+# see https://github.com/napari/napari/issues/5127
+def _modify_layer_gui(viewer):
+    from napari._qt.layer_controls.qt_layer_controls_container import QtLayerControlsContainer
+    from napari._qt.layer_controls.qt_image_controls import QtImageControls
+    from napari._qt.layer_controls.qt_image_controls_base import range_to_decimals
+    from qtpy.QtWidgets import QPushButton
+
+    # search for GUI in napari viewer
+    obj = viewer.window.qt_viewer.dockLayerControls
+    for i in range(obj.layout().count()):
+        if obj.layout().itemAt(i) is not None and obj.layout().itemAt(i).widget is not None:
+            item = obj.layout().itemAt(i).widget()
+            if isinstance(item, QtLayerControlsContainer):
+                layer_controls = item
+                if layer_controls is not None:
+                    obj2 = layer_controls
+                    for i in range(obj2.layout().count()):
+                        if obj2.layout().itemAt(i) is not None:
+                            item = obj2.layout().itemAt(i).widget()
+                            if isinstance(item, QtImageControls):
+                                image_controls = item
+
+                                if image_controls is not None:
+                                    # Modify GUI: Add a "Reset" button to the auto-contrast button bar
+                                    def reset_display_range():
+                                        layer = list(viewer.layers.selection)[0]
+                                        layer.contrast_limits_range = [layer.data.min(), layer.data.max()]
+                                        layer.contrast_limits = list(layer.contrast_limits_range)
+
+                                        contrastLimitsSlider = image_controls.contrastLimitsSlider
+                                        decimals = range_to_decimals(
+                                            layer.contrast_limits_range, layer.dtype
+                                        )
+                                        contrastLimitsSlider.setRange(*layer.contrast_limits_range)
+                                        contrastLimitsSlider.setSingleStep(10 ** -decimals)
+                                        contrastLimitsSlider.setValue(layer.contrast_limits)
```

Comment:
why you write this instead of `layer.reset_contrast_limits_range()` and `layer.reset_contrast_limits()`.  

contrast `contrastLimitsSlider` reseting is connected to update layer `contrast_limits_range`

Reply:
I tried that and it didn't work

Reply:
Hm. This looks like a bug.